hybrid/compartment_examples/inter_comp_call/clean.s

```diff
@@ -0,0 +1,49 @@
+// Inner helper for cleaning capabilities from registers, either side of an
```

Comment:
These `.s` files are now an indirect dependency in the `Makefile`, since they are included by `switch_compartment.S`, and they are essentially "hidden" from the compilation process. While I appreciate the de-duplication, perhaps there is a better way to go around this, rather than hiding this process from the reader...

Reply:
My understanding is they can't really be compiled separately as they need to in the `[switch_compartment, switch_compartment_end)` address range, which is why they're `#include`d in `switch_compartment.S`. Also, plenty of other `.S` files `#include` other assembly sources, so perhaps you are suggesting this is a more general problem in this project?

Reply:
> as they need to in the [switch_compartment, switch_compartment_end) address range,

Right, I think I forgot that this example tries to do some execution compartmentalization (I did forget)? If that is the case, then this make sense, and they can't just be stand-alone files.

> Also, plenty of other .S files

Perhaps. I didn't appreciate the difference between `.s` and `.S` when I initially started working in this repo, so it might be worth considering. On the other hand, if it works, don't fix it.

Reply:
> Perhaps. I didn't appreciate the difference between .s and .S when I initially started working in this repo, so it might be worth considering. On the other hand, if it works, don't fix it.

Oh, I personally don't mind unnecessary uses of `.S` (`.S` just means "run the pre-processor on this file"). What I meant to say is there are other `.s``/.S` files in this project that are "hidden" from the compilation process (i.e. they are inlined in a main `.s`/`.S` rather than compiled individually).

Reply:
Right, I guess once we do a major refactoring, we can maybe care if this is a problem or not. For now, this change is fine. The other part, regarding the location of `clean` needing to be within the specified range, if this is due to execution compartmentalization, then we can leave this as is.

Reply:
Yeah, `clean` needs to be within the bounds of the PCC of the compartment switcher (otherwise calling `clean` would trigger a CHERI protection violation).